### PR TITLE
2424 ts1 parsing

### LINKED ISSUE #2424: Complete TANF Section 1 Parsing
**Description:**
As follow-on to #132 , we need to complete content parsing for T2 and T3 models for TANF Section 1.

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] Record types T2 and T3 can be parsed into their associated models.
- [ ] Relevant category 1 errors for these record types are implemented.
- [ ] Models/Documents are updated to named spec provided by @ADPennington 
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [x] Adjust T2/T3 field names per OFA spec
- [x] Implement parsing in vein of T1 using schema_defs format
- [x] Update `tanf_parser.parse` to handle new parsing functions for these record types
- [x] Refactor/abstract as is convenient for duplicative logic
- [x] Run Testing Checklist and confirm all tests pass

**Notes:**
DISAGGREGATED TANF DATA COLLECTION
TRANSMISSION FORMAT SECTION 1  
https://www.acf.hhs.gov/sites/default/files/documents/ofa/tanf_data_report_section1_10_2008.pdf 
https://www.acf.hhs.gov/ofa/policy-guidance/final-tanf-ssp-moe-data-reporting-system-transmission-files-layouts-and-edits 

**Supporting Documentation:**

**Open Questions:**

### PR DESCRIPTION
**Pull request by elipe17**
## Summary of Changes
- Added schema for TANF T2/T3
- Added unit tests for T2/T3 submissions
Pull request closes #2424 

## How to Test
```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up
```
1. Open http://localhost:3000/ and sign in.
1. Navigate to Data files and upload one of the test files included in this PR
1. Ensure parsing passes and errors logged in the console match what is expected. 

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] Record types T2 and T3 can be parsed into their associated models.
+ [x] Relevant category 1 errors for these record types are implemented.
+ [x] Models/Documents are updated to named spec provided by @ADPennington 
+ [x] Testing Checklist has been run and all tests pass
+ [x] README is updated, if necessary
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Comment by elipe17** [comment]
For the schema field names I literally copy and pasted the values. Some of the fields have meta data like 'ITEM66A' prepended onto them as per the name in the spec. Do we prefer to remove that meta in favor of some other readable string?

Also, the number of migrations is a drag but I could not get the field changes/renames to work any other way. Would love if someone knows a better way to get them consolidated without having to impute default values.

File: tdrs-backend/tdpservice/parsers/tanf_parser.py [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
 
-def active_t1_parser(line, line_number):
-    """Parse line in datafile as active case data, T1 only."""
-    family_case_schema = t1_schema()
-    # create search_index model
-    t1 = TANF_T1()
+def active_parse(actual_line_length, min_line_length, schema, line, line_number, model):
+    """Parse line in datafile as active case data."""
```
Comment (jtimpe) [inline tdrs-backend/tdpservice/parsers/tanf_parser.py]: not sure how this file ended up in your branch but it should no longer exist

File: tdrs-backend/tdpservice/search_indexes/models/tanf.py [hunk 2]
```diff
@@ -75,77 +75,79 @@ class TANF_T2(models.Model):
...
+    DOB = models.IntegerField(null=False, blank=False)
+    SSN = models.CharField(max_length=9, null=False, blank=False)
+    ITEM34A_HISPANIC_OR_LATINO = models.CharField(max_length=1, null=False, blank=False)
+    ITEM34B_AMERICAN_INDIAN_OR_ALASKA_NATIVE = models.CharField(max_length=1, null=False, blank=False)
+    ITEM34C_ASIAN = models.CharField(max_length=1, null=False, blank=False)
+    ITEM34D_BLACK_OR_AFRICAN_AMERICAN = models.CharField(max_length=1, null=False, blank=False)
+    ITEM34E_NATIVE_HAWAIIAN_OR_OTHER_PACIFIC_ISLANDER = models.CharField(max_length=1, null=False, blank=False)
+    ITEM34F_WHITE = models.CharField(max_length=1, null=False, blank=False)
+    GENDER = models.IntegerField(null=False, blank=False)
+    ITEM36A_RECEIVES_FEDERAL_DISABILITY_INSURANCE_OASDI_PROGRAM = models.CharField(max_length=1, null=False,
```
Comment (jtimpe) [inline tdrs-backend/tdpservice/search_indexes/models/tanf.py]: i'm thinking we can probably drop the `ITEM36A` etc from the item name

File: tdrs-backend/tdpservice/search_indexes/migrations/0008_auto_20230424_1913.py [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('search_indexes', '0007_ssp_m1_ssp_m2_ssp_m3'),
+    ]
+
+    operations = [
```
Comment (jtimpe) [inline tdrs-backend/tdpservice/search_indexes/migrations/0008_auto_20230424_1913.py]: i'm getting a migration issue locally. `makemigrations` or `migrate` is prompting me for a field rename, so `docker-compose up` fails
```bash
Did you rename tanf_t2.aid_aged_blind to tanf_t2.AMOUNT_OF_EARNED_INCOME (a IntegerField)? [y/N]
```

i don't see any `RemoveField` for `aid_aged_blind`. anyway, i think it'd be nice to squash the migrations. in this case, we don't have any `t2/t3` documents we need to preserve so i think you could do the migration in two steps: `RemoveField` for all the old fields, then `AddField` for all the new ones. If delete all migrations, stash your changes, comment out all the old fields, that will allow you to generate the first migration. then you can unstash your changes and generate another.

hopefully that makes all this a little simpler and more predictable

**Review by jtimpe — CHANGES_REQUESTED** [review]

File: tdrs-backend/tdpservice/parsers/tanf_parser.py [hunk 4]
```diff
@@ -1,31 +1,23 @@
...
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.DEBUG)
 
-def active_t1_parser(line, line_number):
-    """Parse line in datafile as active case data, T1 only."""
-    family_case_schema = t1_schema()
-    # create search_index model
-    t1 = TANF_T1()
+def active_parse(actual_line_length, min_line_length, schema, line, line_number, model):
```
Comment (raftmsohani) [inline tdrs-backend/tdpservice/parsers/tanf_parser.py]: Nit: change "model" to category_instance or maybe something more readable. "model" is very generic

File: tdrs-backend/tdpservice/parsers/tanf_parser.py [hunk 5]
```diff
@@ -1,31 +1,23 @@
...
     if actual_line_length < min_line_length:
         logger.error('Expected minimum line length of %s, got: %s', min_line_length, actual_line_length)
         return
 
-    for field in family_case_schema.get_all_fields():
-        if field.name == 'blank':
-            # We are discarding this data.
-            break
+    for field in schema.get_all_fields():
         content = line[field.start-1:field.end]  # descriptor pdfs were off by one, could also adjust start values
```
Comment (raftmsohani) [inline tdrs-backend/tdpservice/parsers/tanf_parser.py]: Remove line comment

File: tdrs-backend/tdpservice/parsers/tanf_parser.py [hunk 6]
```diff
@@ -43,15 +35,25 @@ def active_t1_parser(line, line_number):
...
+    # model.full_clean()
+    model.save()
+
+def active_t1_parser(line, line_number):
+    """Parse line in datafile as active case data, T1 only."""
+    family_case_schema = t1_schema()
+    # create search_index model
+    t1 = T1()
+
+    min_line_length = 118  # we will need to adjust for other types
```
Comment (raftmsohani) [inline tdrs-backend/tdpservice/parsers/tanf_parser.py]: Add a constant file and define 118 in that file, something like: MIN_PARSE_LINE_LENGTH = 118

File: tdrs-backend/tdpservice/parsers/tanf_parser.py [hunk 7]
```diff
@@ -67,6 +69,26 @@ def active_t1_parser(line, line_number):
         )
     '''
 
+def active_t2_parser(line, line_number):
+    """Parse line in datafile as active case data, T1 only."""
+    adult_data_schema = t2_schema()
+    # create search_index model
+    t2 = T2()
+
+    min_line_length = 156  # we will need to adjust for other types
```
Comment (raftmsohani) [inline tdrs-backend/tdpservice/parsers/tanf_parser.py]: Same to my other comment, add this to a constant file

File: tdrs-backend/tdpservice/parsers/tanf_parser.py [hunk 8]
```diff
@@ -67,6 +69,26 @@ def active_t1_parser(line, line_number):
...
+    actual_line_length = len(line)
+    active_parse(actual_line_length, min_line_length, adult_data_schema, line, line_number, t2)
+
+def active_t3_parser(line, line_number):
+    """Parse line in datafile as active case data, T1 only."""
+    children_data_schema = t3_schema()
+    # create search_index model
+    t2 = T3()
+
+    min_line_length = 156  # we will need to adjust for other types
```
Comment (raftmsohani) [inline tdrs-backend/tdpservice/parsers/tanf_parser.py]: Same to my other comment, add this to a constant file

File: tdrs-backend/tdpservice/parsers/test/test_parse.py [hunk 9]
```diff
@@ -85,13 +85,15 @@ def test_big_file(stt_user, stt):
 @pytest.mark.django_db
 def test_parse_big_file(test_big_file):
     """Test parsing of ADS.E2J.FTP1.TS06."""
-    expected_errors_count = 1828
     expected_t1_record_count = 815
-
+    expected_t2_record_count = 882
```
Comment (raftmsohani) [inline tdrs-backend/tdpservice/parsers/test/test_parse.py:89]: Nit: use capital letters to define constants: EXPECTED....

**Review by raftmsohani — CHANGES_REQUESTED** [review]
Apply my comments

File: tdrs-backend/tdpservice/parsers/schema_defs/tanf/t3.py [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+child_two = RowSchema(
+    model=TANF_T3,
+    quiet_preparser_errors=True,
+    preparsing_validators=[
+        validators.notEmpty(start=60, end=101),
+    ],
+    postparsing_validators=[],
+    fields=[
+        Field(name='RecordType', type='string', startIndex=0, endIndex=2, required=True, validators=[
+        ]),
```
Comment (andrew-jameson) [inline tdrs-backend/tdpservice/parsers/schema_defs/tanf/t3.py]: Nitpick: Not sure if this is a side-effect of linting but we have inconsistent newlines between the brackets of `validators=[]`. Would prefer this to be consistent with how we're going about it in #2479 which doesn't split the line but looks like this:

```
   fields=[
        Field(name='RecordType', type='string', startIndex=0, endIndex=2,
              required=True, validators=[]),
        Field(name='RPT_MONTH_YEAR', type='number', startIndex=2, endIndex=8,
              required=True, validators=[]),
             ```

**Review by jtimpe — APPROVED** [review]
agree with andrew's comment on formatting of the schema files, but everything else looks good!

**Review by raftmsohani — APPROVED** [review]
LGTM!

**Review by andrew-jameson — APPROVED** [review]

File: tdrs-backend/tdpservice/parsers/schema_defs/tanf/t2.py [hunk 11]
```diff
@@ -0,0 +1,129 @@
+"""Schema for HEADER row of all submission types."""
+
+
+from ...util import RowSchema, Field
+from ... import validators
+from tdpservice.search_indexes.models.tanf import TANF_T2
+
+
+t2 = RowSchema(
```
Comment (ADPennington) [inline tdrs-backend/tdpservice/parsers/schema_defs/tanf/t2.py:9]: @elipe17 this is the list of fields for TANF T2 records:

```
T2col_list = ['RecordType','RPT_MONTH_YEAR', 'CASE_NUMBER','FAMILY_AFFILIATION','NONCUSTODIAL_PARENT', 'DATE_OF_BIRTH','SSN','RACE_HISPANIC',
            'RACE_AMER_INDIAN','RACE_ASIAN',  'RACE_BLACK',  'RACE_HAWAIIAN', 'RACE_WHITE',
            'GENDER', 'FED_OASDI_PROGRAM',  'FED_DISABILITY_STATUS', 'DISABLED_TITLE_XIVAPDT', 'AID_AGED_BLIND',                
            'RECEIVE_SSI','MARITAL_STATUS', 'RELATIONSHIP_HOH',  'PARENT_MINOR_CHILD',  'NEEDS_PREGNANT_WOMAN',          
            'EDUCATION_LEVEL', 'CITIZENSHIP_STATUS', 'COOPERATION_CHILD_SUPPORT', 'MONTHS_FED_TIME_LIMIT',  'MONTHS_STATE_TIME_LIMIT',       
            'CURRENT_MONTH_STATE_EXEMPT', 'EMPLOYMENT_STATUS',  'WORK_ELIGIBLE_INDICATOR',   'WORK_PART_STATUS',  'UNSUB_EMPLOYMENT',              
            'SUB_PRIVATE_EMPLOYMENT',  'SUB_PUBLIC_EMPLOYMENT',  'WORK_EXPERIENCE_HOP', 'WORK_EXPERIENCE_EA', 'WORK_EXPERIENCE_HOL',           
            'OJT',  'JOB_SEARCH_HOP',  'JOB_SEARCH_EA', 'JOB_SEARCH_HOL',  'COMM_SERVICES_HOP',             
            'COMM_SERVICES_EA', 'COMM_SERVICES_HOL', 'VOCATIONAL_ED_TRAINING_HOP', 'VOCATIONAL_ED_TRAINING_EA', 'VOCATIONAL_ED_TRAINING_HOL',    
            'JOB_SKILLS_TRAINING_HOP',  'JOB_SKILLS_TRAINING_EA',  'JOB_SKILLS_TRAINING_HOL',  'ED_NO_HIGH_SCHOOL_DIPL_HOP',  'ED_NO_HIGH_SCHOOL_DIPL_EA',     
            'ED_NO_HIGH_SCHOOL_DIPL_HOL', 'SCHOOL_ATTENDENCE_HOP', 'SCHOOL_ATTENDENCE_EA',   'SCHOOL_ATTENDENCE_HOL',   'PROVIDE_CC_HOP',                
            'PROVIDE_CC_EA',  'PROVIDE_CC_HOL',  'OTHER_WORK_ACTIVITIES',  'DEEMED_HOURS_FOR_OVERALL',  'DEEMED_HOURS_FOR_TWO_PARENT',   
            'EARNED_INCOME', 'UNEARNED_INCOME_TAX_CREDIT', 'UNEARNED_SOCIAL_SECURITY', 'UNEARNED_SSI', 'UNEARNED_WORKERS_COMP',         
            'OTHER_UNEARNED_INCOME']
```
also located [here](https://gorafttech.sharepoint.com/:u:/r/sites/TDRSResearchDesign/Shared%20Documents/Dev/dev/Parsing/py/[TANF_Parsing.py](https://gorafttech.sharepoint.com/:u:/r/sites/TDRSResearchDesign/Shared%20Documents/Dev/dev/Parsing/py/TANF_Parsing.py?csf=1&web=1&e=qFRqo9)?csf=1&web=1&e=qFRqo9)

File: tdrs-backend/tdpservice/parsers/schema_defs/tanf/t3.py [hunk 12]
```diff
@@ -0,0 +1,91 @@
+"""Schema for HEADER row of all submission types."""
+
+
+from ...util import MultiRecordRowSchema, RowSchema, Field
+from ... import validators
+from tdpservice.search_indexes.models.tanf import TANF_T3
+
+
+child_one = RowSchema(
+    model=TANF_T3,
```
Comment (ADPennington) [inline tdrs-backend/tdpservice/parsers/schema_defs/tanf/t3.py:10]: @elipe17 see [link](https://gorafttech.sharepoint.com/:u:/r/sites/TDRSResearchDesign/Shared%20Documents/Dev/dev/Parsing/py/[TANF_Parsing.py](https://gorafttech.sharepoint.com/:u:/r/sites/TDRSResearchDesign/Shared%20Documents/Dev/dev/Parsing/py/TANF_Parsing.py?csf=1&web=1&e=qFRqo9)?csf=1&web=1&e=qFRqo9) to T3 record field names.

**Review by ADPennington — CHANGES_REQUESTED** [review]
@elipe17 this is in good shape, but please note:

- please verify that the T2 and T3 record field names are correct. 
- there seems to be an issue when submitting more than one file as well as when only one large file is submitted (testing notes below) (all parsing files and test data files are in raft's sharepoint [site](https://gorafttech.sharepoint.com/:f:/r/sites/TDRSResearchDesign/Shared%20Documents/Dev/dev/[Parsing](https://gorafttech.sharepoint.com/:f:/r/sites/TDRSResearchDesign/Shared%20Documents/Dev/dev/Parsing?csf=1&web=1&e=fIWkpC)?csf=1&web=1&e=fIWkpC)

---

### testing notes

- Submitting 1 `small_tanf_section1.txt` for TANF section 1 results in `Successfully submitted section(s): 1 on Fri May 12 2023` :heavy_check_mark:
  - T1 records added to DAC
  - T2 records added to DAC
  - T3 records with one child added to DAC
  - T3 records with 2 children added to DAC
  - data submission email received
  - files successfully transferred to TDRS. 

- Submitting 2 or more `small_tanf_section1.txt` (one file per section) results in `Request failed with status code 429` upon submission :warning:

    ```
    13:21:15.242: [APP/PROC/WEB.0] 2023/05/12 17:21:15 [error] 114#0: *941 limiting requests, excess: 0.600 by zone "limitreqsbyaddr", client: 2601:140:8c00:55b:31a8:1c25:e502:41c7, server: , request: "POST /v1/data_files/ HTTP/1.1", host: "tdp-frontend-qasp.app.cloud.gov", referrer: "https://tdp-frontend-qasp.app.cloud.gov/data-files"
    13:21:15.244: [APP/PROC/WEB.0] 2023/05/12 17:21:15 [error] 114#0: *942 limiting requests, excess: 0.500 by zone "limitreqsbyaddr", client: 2601:140:8c00:55b:31a8:1c25:e502:41c7, server: , request: "POST /v1/data_files/ HTTP/1.1", host: "tdp-frontend-qasp.app.cloud.gov", referrer: "https://tdp-frontend-qasp.app.cloud.gov/data-files"
    ```
- 15MB TANF Section 1 files result in `Request failed with status code 502` upon submission :warning:
    ```
    13:19:00.839: [APP/PROC/WEB.0] 2023/05/12 17:19:00 [error] 114#0: *929 client intended to send too large body: 15477701 bytes, client: 2601:140:8c00:55b:31a8:1c25:e502:41c7, server: , request: "POST /v1/data_files/ HTTP/1.1", host: "tdp-frontend-qasp.app.cloud.gov", referrer: "https://tdp-frontend-qasp.app.cloud.gov/data-files"
    ```

### other testing

- as expected:
  - `TANF Section 1` data parsing implemented when user's STT type is `Territory`
  - data parsing not yet implemented for `TANF Sections 1, 2, 3, and 4` when user's STT type is `Tribe` 
  - data parsing not yet implemented for `TANF Sections 2, 3, and 4` 
  - data parsing not yet implemented for `SSP Sections 1, 2, 3, and 4`

**Comment by ADPennington** [comment]
@elipe17 can you resolve the conflicts here?

**Comment by ADPennington** [comment]
@elipe17 can you look into the traceback below. appears to be a migration-related issue. deployed to qasp env. 

```
08:40:17.834: [APP/PROC/WEB.0] Running migrations:
08:40:17.845: [APP/PROC/WEB.0] Traceback (most recent call last):
08:40:17.845: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
08:40:17.845: [APP/PROC/WEB.0] return self.cursor.execute(sql, params)
08:40:17.845: [APP/PROC/WEB.0] psycopg2.errors.DuplicateColumn: column "CASE_NUMBER" of relation "search_indexes_tanf_t2" already exists
08:40:17.845: [APP/PROC/WEB.0] The above exception was the direct cause of the following exception:
08:40:17.845: [APP/PROC/WEB.0] Traceback (most recent call last):
08:40:17.845: [APP/PROC/WEB.0] File "/home/vcap/app/manage.py", line 31, in <module>
08:40:17.845: [APP/PROC/WEB.0] main()
08:40:17.845: [APP/PROC/WEB.0] File "/home/vcap/app/manage.py", line 27, in main
08:40:17.845: [APP/PROC/WEB.0] execute_from_command_line(sys.argv)
08:40:17.845: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
08:40:17.845: [APP/PROC/WEB.0] utility.execute()
08:40:17.845: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
08:40:17.845: [APP/PROC/WEB.0] self.fetch_command(subcommand).run_from_argv(self.argv)
08:40:17.846: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
08:40:17.846: [APP/PROC/WEB.0] self.execute(*args, **cmd_options)
08:40:17.846: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
08:40:17.846: [APP/PROC/WEB.0] output = self.handle(*args, **options)
08:40:17.846: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/core/management/base.py", line 89, in wrapped
08:40:17.846: [APP/PROC/WEB.0] res = handle_func(*args, **kwargs)
08:40:17.846: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/core/management/commands/migrate.py", line 244, in handle
08:40:17.846: [APP/PROC/WEB.0] post_migrate_state = executor.migrate(
08:40:17.846: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/migrations/executor.py", line 117, in migrate
08:40:17.846: [APP/PROC/WEB.0] state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
08:40:17.846: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
08:40:17.846: [APP/PROC/WEB.0] state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
08:40:17.846: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
08:40:17.846: [APP/PROC/WEB.0] state = migration.apply(state, schema_editor)
08:40:17.846: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/migrations/migration.py", line 126, in apply
08:40:17.846: [APP/PROC/WEB.0] operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
08:40:17.846: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/migrations/operations/fields.py", line 104, in database_forwards
08:40:17.846: [APP/PROC/WEB.0] schema_editor.add_field(
08:40:17.846: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/base/schema.py", line 522, in add_field
08:40:17.847: [APP/PROC/WEB.0] self.execute(sql, params)
08:40:17.847: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/base/schema.py", line 145, in execute
08:40:17.847: [APP/PROC/WEB.0] cursor.execute(sql, params)
08:40:17.847: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 98, in execute
08:40:17.847: [APP/PROC/WEB.0] return super().execute(sql, params)
08:40:17.847: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 66, in execute
08:40:17.847: [APP/PROC/WEB.0] return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
08:40:17.847: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
08:40:17.847: [APP/PROC/WEB.0] return executor(sql, params, many, context)
08:40:17.847: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 79, in _execute
08:40:17.847: [APP/PROC/WEB.0] with self.db.wrap_database_errors:
08:40:17.847: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/utils.py", line 90, in __exit__
08:40:17.847: [APP/PROC/WEB.0] raise dj_exc_value.with_traceback(traceback) from exc_value
08:40:17.847: [APP/PROC/WEB.0] File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
08:40:17.847: [APP/PROC/WEB.0] return self.cursor.execute(sql, params)
08:40:17.847: [APP/PROC/WEB.0] django.db.utils.ProgrammingError: column "CASE_NUMBER" of relation "search_indexes_tanf_t2" already exists
```

**Review by ADPennington — APPROVED** [review]
> @elipe17 this is in good shape, but please note:
> 
> * please verify that the T2 and T3 record field names are correct.
> * there seems to be an issue when submitting more than one file as well as when only one large file is submitted (testing notes below) (all parsing files and test data files are in raft's sharepoint [site](https://gorafttech.sharepoint.com/:f:/r/sites/TDRSResearchDesign/Shared%20Documents/Dev/dev/%5BParsing%5D(https://gorafttech.sharepoint.com/:f:/r/sites/TDRSResearchDesign/Shared%20Documents/Dev/dev/Parsing?csf=1&web=1&e=fIWkpC)?csf=1&web=1&e=fIWkpC)
> 

@elipe17 

- field names look good now!
- intermittent failures with larger files. By the way, in case its helpful for #2532 i captured the following 400 response when uploading files above and just under 50MB cap. I got down to 47MB and it still failed cc: @andrew-jameson 

2424p1

```
09:13:23.736: [RTR.2] tdp-frontend-qasp.app.cloud.gov - [2023-05-24T13:13:23.268568861Z] "POST /v1/data_files/ HTTP/1.1" 400 49918073 128 "https://tdp-frontend-qasp.app.cloud.gov/data-files" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36" "127.0.0.1:9144" "10.10.1.41:61046" x_forwarded_for:"52.86.158.198, 127.0.0.1" x_forwarded_proto:"https" vcap_request_id:"6b0888aa-86c7-4e5f-7d37-09c0d9053afb" response_time:0.467467 gorouter_time:0.000129 app_id:"66b86b25-3c40-48a9-916e-c5319e51f308" app_index:"0" instance_id:"e423ba5b-53eb-4c27-4e9d-4d2b" x_cf_routererror:"-" x_b3_traceid:"2297f895ab7753f565988e9219f5ae30" x_b3_spanid:"65988e9219f5ae30" x_b3_parentspanid:"-" b3:"2297f895ab7753f565988e9219f5ae30-65988e9219f5ae30"
```

**Pull request merged**